addons/account/models/account_move.py

```diff
@@ -605,9 +612,45 @@ def _onchange_invoice_vendor_bill(self):
             self.invoice_vendor_bill_id = False
             self._recompute_dynamic_lines()
 
+    def _calculate_untaxed_price(self, tax):
+        self.ensure_one()
+        is_refund = self.move_type in ('out_refund', 'in_refund')
+        tax_amount = tax.amount * tax._get_tax_repartition_lines_sum_factor(is_refund)
+        if tax.price_include:
+            return self.quick_edit_total_amount
+        if tax.amount_type == 'fixed':
+            return self.quick_edit_total_amount - tax_amount
+        elif tax.amount_type == 'percent':
+            return self.quick_edit_total_amount / (1 + (tax_amount / 100.0))
+        elif tax.amount_type == 'division':
+            return self.quick_edit_total_amount * (1 - (tax_amount / 100.0))
+        return self.quick_edit_total_amount
+
+    @api.onchange('quick_edit_total_amount')
+    def _onchange_quick_edit_total_amount(self):
+        if not self.quick_edit_total_amount:
+            return
+        self.line_ids = [Command.clear()]
+        account_id = self.journal_id.default_account_id.id
+        if self.partner_id:
+            account_id = self.env['account.account']._order_by_frequency_per_partner(self.company_id.id, self.partner_id.id, self.move_type, limit=1)[0]
+        tax = self.company_id.account_sale_tax_id if self.journal_id.type == 'sale' else self.company_id.account_purchase_tax_id
+        tax = self.fiscal_position_id.map_tax(tax)
+        price_untaxed = self._calculate_untaxed_price(tax)
```

Comment:
No need for `_calculate_untaxed_price`
```suggestion
        price_untaxed = tax.compute_all(self.quick_edit_total_amount)['total_excluded']
```